schemas/2016-03-30/Microsoft.ContainerService.json

```diff
@@ -0,0 +1,229 @@
+      "required": [
+        "id"
+      ]
+    },
+    "orchestratorProfile": {
+      "properties": {
+        "orchestratorType": {
+          "type": "string"
+        }
+      },
+      "type": "object",
+      "required": [
+        "orchestratorType"
+      ]
+    },
+    "masterProfile": {
+      "properties": {
+        "count": {
+          "oneOf": [
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            },
+            {
+              "type": "integer"
+            }
+          ]
+        },
+        "dnsPrefix": {
+          "type": "string"
+        }
+      },
+      "type": "object",
+      "required": [
+        "count",
+        "dnsPrefix"
+      ]
+    },
+    "publicKey": {
+      "properties": {
+        "keyData": {
+          "type": "string"
+        }
+      },
+      "type": "array",
+      "required": [
+        "keyData"
+      ]
+    },
+    "ssh": {
+      "properties": {
+        "publicKeys": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/publicKey"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ]
+        }
+      },
+      "type": "object",
+      "required": [
+        "publicKeys"
+      ]
+    },
+    "linuxProfile": {
+      "properties": {
+        "adminUsername": {
+          "type": "string"
+        },
+        "ssh": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/ssh"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ]
+        }
+      },
+      "type": "object",
+      "required": [
+        "adminUsername",
+        "ssh"
+      ]
+    },
+    "agentPoolProfile": {
+      "properties": {
+        "name": {
+          " type": "string"
+        },
+        "count": {
+          "oneOf": [
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            },
+            {
+              "type": "integer"
+            }
+          ]
+        },
+        "vmSize": {
+          "type": "string"
+        },
+        "dnsPrefix": {
+          "type": "string"
+        }
+      },
+      "type": "object",
+      "required": [
+        "name",
+        "count",
+        "vmSize",
+        "dnsPrefix"
+      ]
+    },
+    "agentPoolProfiles": {
+      "items": {
+        "oneOf": [
+          {
+            "$ref": "#/definitions/agentPoolProfile"
+          },
+          {
+            "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+          }
+        ]
+      },
+      "type": "array",
+      "required": [
+        "items"
```

Comment:
No. AgentPoolProfiles is basically an array with different properties (specified in agentPoolProfile).